Eq 391 pipe an answer

weapdiv-david opened the pull request:
### What is the context of this PR?

This pull request allows an author to specify an alias for a response, and then refer to that response in the text of a later question.

The star wars kitchen sink schema has been modified to demonstrate this functionality
### How to review

1) Check out the branch
2) Run the tests
3) Run the app and go to the dev console
4) Select the star wars survey
5) Answer the question about Chewbaccas age and proceed through the questionnaire
6) On the last page of the questionnaire, verify that there is an additional question asking you to confirm Chewbaccas age and that the value you entered is contained within the question text
- [x] Do all commits have sensible commit messages?
- [x] Is all new code unit tested?
- [x] Are there integration tests if needed?
- [x] Is there sufficient logging?
- [x] Is there documentation or is the code self-describing?
### Who worked on the PR

@weapdiv-david worked on this PR so it needs to be reviewed by someone other than him

tafftoo committed:
Rename variables and add comments for clearer code

tafftoo committed:
Basic Support for piping a previous response

warrenbailey commented on app/renderer/renderer.py:
```diff
@@ -195,3 +174,52 @@ def _augment_item(self, item, global_errors, global_warnings):
+        end_date = None
+        employment_date = None
+        return_by = None
+
```
I think we can remove the try/except from this, it's not needed anymore. 

While you're at it might as well remove the use of the getters and just use the property

warrenbailey commented on app/model/questionnaire.py:
```diff
@@ -38,6 +40,13 @@ def register(self, item):
 
         self.items_by_id[item.id] = item
 
+        # Register the alias if the item has one
```
This could be extracted to a _register_alias method?

warrenbailey commented:
A couple of minor comments on code structure.

I can't see any unit or integration tests? Are you missing a commit?

weapdiv-david commented:
Refactored to take on your comments and rebased on answer/response commit.

As for tests, there are no new unit tests for this functionality, but it is covered by a modified star wars integration test

warrenbailey commented:
But the modified star wars test isn't in this commit?

warrenbailey commented:
Apparently the story states Lewis must be happy, over to you @LJBabbage
